Don't consider untagged nodes for geometric relations by default

patrickbr opened the pull request:
Since #101, the geometries of untagged nodes are written by default. This adds around 10 billion triples for OSM Planet, which is fine. However, considering these nodes for the geometric relations (`contains`, `intersects`, ...) adds an enormous number of triples. It is also not necessary since the information which node is a member of which way can be obtained via #111. 

Therefore, the default is now to *not* consider untagged nodes for the geometric relations, but to output the triples that express which node is a member of which way. The triples that express which way is a member of which relation were already output by default before.

To change the new default behavior, there are now the options `--add-untagged-nodes-geometric-relations` (to consider untagged nodes for the geometric relations, despite the enormous number of triples this incurs) and `--no-member-triples` (to *not* output the triples that express which node is a member of which way and which way is a member of which relation).

patrickbr commented:
I also updated `libspatialjoin`: `std::runtime_error`s thrown in spatialjoin's worker threads are now clearly printed to `std::err` and `std::exit(1)` is called.

hannahbast reviewed (changes requested):
Thanks a lot for this. Looks fine to me and it has already been tried and tested end to end for OHM and OSM. But I have a suggestion:

1. The number of triples added when not using this option is enormous (also for moderately-sized subsets of OSM) and we can now get that information via #111. I therefore suggest to make it the default that geometric relations are *not* computed for the untagged nodes. The option for computing them could then be called `--geometric-relations-for-untagged-nodes`.

3. If we do this, we should make it the default to compute the triples for relation and way members. The option for *not* computing them could then be called `--no-member-triples-for-relations-and-ways`. Note that the current `--add-way-node-order` is a bit of a misnomer because it also adds member triples for relations.

patrickbr commented:
> 1. The number of triples added when not using this option is enormous [...]. I therefore suggest to make it the default that geometric relations are _not_ computed for the untagged nodes.

Agreed.  This is now implemented in 0dd4cc9. For consistency with existing options the option is called `--add-untagged-nodes-geometric-relations`.

> 2. If we do this, we should make it the default to compute the triples for relation and way members. The option for _not_ computing them could then be called `--no-member-triples-for-relations-and-ways`.

Also implemented in 0dd4cc9. The option is now called `--no-member-triples`, the `-for-relations-and-ways` suffix was a bit redundant. 

>  Note that the current `--add-way-node-order` is a bit of a misnomer because it also adds member triples for relations.

It wasn't a misnomer - up until now, the member triples for relation were *always* written, there was no option to disable this.

hannahbast reviewed (approved):
Thanks a lot for the revision. I have tried the default setting many times now for OSM Planet and it worked like a charm. I just revised the title and the description and will merge this now.
